includes/gateway/class-omise-payment-googlepay.php

```diff
@@ -0,0 +1,237 @@
+                'css'         => Omise_Card_Image::get_css(),
+                'default'     => Omise_Card_Image::get_visa_default_display()
+            ),
+            'accept_mastercard' => array(
+                'type'        => 'checkbox',
+                'label'       => Omise_Card_Image::get_mastercard_image(),
+                'css'         => Omise_Card_Image::get_css(),
+                'default'     => Omise_Card_Image::get_mastercard_default_display()
+            ),
+            'accept_jcb' => array(
+                'type'        => 'checkbox',
+                'label'       => Omise_Card_Image::get_jcb_image(),
+                'css'         => Omise_Card_Image::get_css(),
+                'default'     => Omise_Card_Image::get_jcb_default_display()
+            ),
+            'accept_amex' => array(
+                'type'        => 'checkbox',
+                'label'       => Omise_Card_Image::get_amex_image(),
+                'css'         => Omise_Card_Image::get_css(),
+                'default'     => Omise_Card_Image::get_amex_default_display(),
+                'description' => wp_kses(
+                    __('This only controls the allowed card networks GooglePay will allow the customer to select.<br />It is not related to card processing on Omise payment gateway.', 'omise'),
+                    array('br' => array())
+                )
+            )
+        );
+    }
+
+    /**
+     * Register all required javascripts
+     */
+    private function register_omise_googlepay_scripts()
+    {
+        wp_enqueue_script(
+            'googlepay-button-component',
+            plugins_url('../assets/javascripts/googlepay-button-index.umd.min.js', dirname(__FILE__)),
+            array(),
+            WC_VERSION,
+            true
+        );
+    }
+
+    public function get_icon()
+    {
+        return null;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public
+    function payment_fields()
+    {
+        parent::payment_fields();
+        Omise_Util::render_view(
+            'templates/payment/form-googlepay.php',
+            array('config' => $this->google_pay_button_scripts())
+        );
+    }
+
+    private
+    function google_pay_button_scripts()
+    {
+        wp_enqueue_script('omise-js', 'https://cdn.omise.co/omise.js', array('jquery'), WC_VERSION, true);
+
+        $cardNetworks = [];
+        $this->get_option('accept_amex') == 'yes' ? array_push($cardNetworks, "AMEX") : null;
+        $this->get_option('accept_jcb') == 'yes' ? array_push($cardNetworks, "JCB") : null;
+        $this->get_option('accept_mastercard') == 'yes' ? array_push($cardNetworks, "MASTERCARD") : null;
+        $this->get_option('accept_visa') == 'yes' ? array_push($cardNetworks, "VISA") : null;
+        $allowedCardNetworks = json_encode($cardNetworks);
+
+        return array("script" =>
+        "<script type='module'>
+                 const button = document.createElement('google-pay-button')
```

Comment:
The indentation for this html/js section seems not quite consistent.

Reply:
I'm using this extension on vs code https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client should I manually format the js code?

Reply:
Could you show how it looks like in VS Code? Even in the JS code section below the indentations are also not equal 🤔

Reply:
yea you're right. I've reformatted the JS code. please have another look